### ISSUE TITLE
[QUESTION] Which is the best way to easily switch between compiling locally and pushing to githug-pages?

### ISSUE THREAD

**Issue by vncoelho** [mmistakes/minimal-mistakes#1732]
Hi, guys.

First of all, congrulations for this great template.
I am wondered about the possibilities that ideas like Jekyll are promoting.

What is the  best way to switch between local tests and github.
Because if I run `remote_theme` locally I will problems with 'bundle update & bundle exec jekyll serve`

In this sense, I am constantly commenting and uncommenting the gemfile and config.
```ruby
source "https://rubygems.org"

# Remove comments local tests
# gem "jekyll", "~> 3.5"
# gem "minimal-mistakes-jekyll"

# If you have any plugins, put them here!
group :jekyll_plugins do
  gem "jekyll-paginate"
  gem "jekyll-sitemap"
  gem "jekyll-gist"
  gem "jekyll-feed"
  gem "jemoji"
  gem "jekyll-algolia"
end

# Remove comments for github-pages
/#
gem "github-pages", group: :jekyll_plugins
gem "tzinfo-data"
gem "wdm", "~> 0.1.0" if Gem.win_platform?
/#
```

And switching from remote to theme:

```ruby
remote_theme             : "mmistakes/minimal-mistakes"
#theme			 : "minimal-mistakes-jekyll"
```

**Comment by vncoelho** [comment]
I am now using the following script:

```bash
#!/bin/bash
echo "Starting bundle jekyll serve"
cp _config.yml ./_config-temp.yml
sed -i '/remote_theme/c\theme			 : "minimal-mistakes-jekyll"' ./_config-temp.yml
bundle exec jekyll serve --config ./_config-temp.yml
```

Furthermore, I am ignoring the Gemfile since github is being able to generate the page without it.

Is it correct?

**Comment by vncoelho** [comment]
Hey, @mmistakes, thanks very much for this dip attention.

Before closing this issue I am still struggling with something.
I tried your method, but the page did not went on locally, at port 4000.

**Comment by vncoelho** [comment]
Now I noticed that it only worked locally when I added `github: [metadata]`  to the `_config.yml`

**Comment by mmistakes** [comment]
`rake preview` is a development thing. It's used when adding new features and uses the source files in `/example` to spin up `jekyll serve`.

Jekyll's Minima theme did it first... that's where I got the idea.

And yes, remote theme will be slightly slower as it has to downloaded the theme every time it builds. There was some [discussion about caching between builds](https://github.com/benbalter/jekyll-remote-theme/issues/22), so maybe take a look at [that project](https://github.com/benbalter/jekyll-remote-theme) if you have feedback to add.

Closing as these concerns are all Jekyll (and it's dependencies) related... nothing specific here to the theme.
